A new way to get into the Hazardous Waste Sarcophagus

thelonestander opened the pull request:
Adds a book that spawns with the military ID in the Hazardous Waste Sarcophagi.


SUMMARY: Content "Adds a new book that spawns with the military id card in the Hazardous Waste Sarcophagi overmap locations"


#### Purpose of change

1.) The main reason to change this is to give players the ability to go into the location and see what it may have downstairs. Im not spoiling anything, but its cool. 
2.) You can apparently access this location via a questline, however. I feel as though not everyone will find the quest giver (I don't even know who gives the quest) so I gave this as an extra option just in case players cant find it either. 
3.) Another good reason to have this as an option is to allow players to access what IS below and might be able to make use of it. Those of you who have been down to the basement know what loot is there. This may open up options, and make nanofab templets more...useable. 

#### Describe the solution

Added the new book as a dedicated spawn. In game youll ever only need one so this seems like a viable option. 
Also since it ONLY spawns in these locations this gives the people the *want* to loot them as well. 

#### Describe alternatives you've considered

None.

#### Testing

Tested in game, loaded up. Debug spawned the overmap special in and boom it spawned. Have fun guys! 

#### Additional context

![Screenshot (808)](https://user-images.githubusercontent.com/82045140/184683454-11c4e1c1-e2df-46ff-a7c5-89a392e2aa6f.png)

thelonestander commented on data/json/items/book/misc.json:
```diff
@@ -957,5 +957,24 @@
+    "id": "sa_cryptology_key",
+    "type": "BOOK",
+    "name": { "str_sp": "High Level Commercial and Military Cryptology" },
+    "description": "An incredibly dense volume describing different advanced cryptographical methods, their strenghts, and weaknesses. This text would probably be easier to digest with an added decade in Computer Science Academia.",
```
Fixed! Forgot about this. Thank you.

Coolthulhu commented on data/json/items/book/misc.json:
```diff
@@ -957,5 +957,24 @@
+    "id": "sa_cryptology_key",
+    "type": "BOOK",
+    "name": { "str_sp": "High Level Commercial and Military Cryptology" },
+    "description": "An  incredibly  dense  volume  describing  different  advanced  cryptographical  methods,  their strenghts,  and weaknesses.  This  text  would  probably  be  easier  to  digest  with  an  added  decade  in  Computer  Science  Academia.",
```
A whole load of extra spaces in there.

thelonestander replied:
Fixed and added spaces to "name" since its a descriptive string.

olanti-p commented:
```
(json-error)
Json error: data/mods/dda/../../json/items/book/misc.json:965:135: insufficient spaces at this location.  2 required, but only 1 found.
    Suggested fix: insert " "
    At the following position (marked with caret)

    "type": "BOOK",
    "name": { "str_sp": "High  Level  Commercial  and  Military  Cryptology" },
    "description": "An  incredibly dense volume describing different advanced cryptographical methods, their strenghts, and weaknesses.
                                                                                                                                      ^
                                                                                                                                        This text would probably be easier to digest with an added decade in Computer Science Academia.",
    "weight": 3000,
    "volume": "2 L",
```

Bright Night - same as its ancestor, Dark Days Ahead, - uses monospaced fonts for its curses and SDL builds. Monospaced fonts are what typewriters used to use, and one of the conventions for text style typed with a typewriter was to use single space after a word and double space after a dot - for ease of reading. Same convention was adopted by DDA at some point in the past, and is enforced by automatic checks, and these checks were inherited by BN.

To satisfy the build checks, you'll have to use single space after words and double space after dots (after sentence ends, as @KheirFerrum pointed out).

thelonestander commented:
> ```
> (json-error)
> Json error: data/mods/dda/../../json/items/book/misc.json:965:135: insufficient spaces at this location.  2 required, but only 1 found.
>     Suggested fix: insert " "
>     At the following position (marked with caret)
> 
>     "type": "BOOK",
>     "name": { "str_sp": "High  Level  Commercial  and  Military  Cryptology" },
>     "description": "An  incredibly dense volume describing different advanced cryptographical methods, their strenghts, and weaknesses.
>                                                                                                                                       ^
>                                                                                                                                         This text would probably be easier to digest with an added decade in Computer Science Academia.",
>     "weight": 3000,
>     "volume": "2 L",
> ```
> 
> Bright Nights - same as its ancestor, Dark Days Ahead, - uses monospaced fonts for its curses and SDL builds. Monospaced fonts are what typewriters used to use, and one of the conventions for text typed with a typewriter was to use single space after a word and double space after a dot - for ease of reading. Same convention was adopted by DDA at some point in the past, and is enforced by automatic checks, and these checks (and style) were inherited by BN.
> 
> To satisfy the build checks, you'll have to use single space after words and double space after dots (after sentence ends, as @KheirFerrum pointed out).

Oh dang it. Im so sorry. I read what he said wrong. It should be fixed now. Thank you Olanti

olanti-p commented on data/json/items/book/misc.json:
```diff
@@ -957,5 +957,24 @@
+    "id": "sa_cryptology_key",
+    "type": "BOOK",
+    "name": { "str_sp": "High Level Commercial and Military Cryptology" },
+    "description": "An  incredibly dense volume describing different advanced cryptographical methods, their strenghts, and weaknesses.  This text would probably be easier to digest with an added decade in Computer Science Academia.",
```
```suggestion
    "description": "An incredibly dense volume describing different advanced cryptographical methods, their strenghts, and weaknesses.  This text would probably be easier to digest with an added decade in Computer Science Academia.",
```
Leftover space

Coolthulhu commented:
Wait, so this just adds a high skill computer book?

The description you provided makes it sound like something completely different.

thelonestander commented:
> Wait, so this just adds a high skill computer book?
> 
> The description you provided makes it sound like something completely different.

Yeah, it adds a book to the Sarcophagi so the player can unlock the console and go down. In game you only need one of these codes to access all of them. Thats a vanilla thing. So I just added it to the location so players can go down without having to mine a bunch.

Coolthulhu commented:
> In game you only need one of these codes to access all of them.

Oh, so the book is hardcoded to let you into the sarcophagus?

thelonestander commented:
> > In game you only need one of these codes to access all of them.
> 
> Oh, so the book is hardcoded to let you into the sarcophagus?

I do believe so, the book doesn't just grant access, it gives you a recipe to craft the code to get it. So Youd still have to do a bit of work to get in. So, find book, craft code viola! Access to all the sarcophagi's!

Coolthulhu commented:
> it gives you a recipe to craft the code to get it

Well, it doesn't do that bit.

thelonestander commented:
> > it gives you a recipe to craft the code to get it
> 
> Well, it doesn't do that bit.

It should have. I fixed it, added the recipe. So you should now have to get the book, then craft the code.
